src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/InternalFilterLongReadAlignmentsSAMSpark.java

```diff
@@ -0,0 +1,328 @@
+     * {@link #computeScoreOfConfiguration(List, Set, int)}.
+     *
+     * @param longReads     long read alignments
+     * @param header        header for the long reads
+     * @param toolLogger    logger for, most likely, debugging uses
+     *
+     * @return              contigs with alignments filtered and custom formatted as {@link AlignmentInterval}
+     */
+    static JavaRDD<AlignedContig> filterByScore(final JavaRDD<GATKRead> longReads,
+                                                final SAMFileHeader header,
+                                                final Logger toolLogger) {
+
+        toolLogger.info( "Processing these many raw alignments: " + longReads.count() );
+
+        final JavaRDD<AlignedContig> parsedContigAlignments
+                = new SAMFormattedContigAlignmentParser(longReads, header, false, toolLogger)
+                .getAlignedContigs()
+                .filter(InternalFilterLongReadAlignmentsSAMSpark::contigFilter);
+
+        toolLogger.info( "Primitive filtering based purely on MQ left these many contigs: " + parsedContigAlignments.count() );
+
+        return filterAndSplitGappedAI(parsedContigAlignments);
+    }
+
+    /**
+     * Idea is to keep mapped contig that has at least two alignments over MQ 20,
+     * or in the case of a single alignment, it must be MQ>20.
+     */
+    private static boolean contigFilter(final AlignedContig contig) {
+        if (contig.alignmentIntervals.size() < 2 ) {
+            return (!contig.alignmentIntervals.isEmpty()) && contig.alignmentIntervals.get(0).mapQual > 20;
+        } else {
+            return contig.alignmentIntervals.stream().mapToInt(ai -> ai.mapQual).filter(mq -> mq > 20).count() > 1;
+        }
+    }
+
+    /**
+     * Manages configuration score-based filtering,
+     * then split the gapped alignments by delegating to {@link GappedAlignmentSplitter}.
+     * Note that this step is essentially a flatMap operation, meaning that one contig may return 1 or multiple contigs:
+     *  *) when 1 contig is yield, it means the contig has only 1 configuration that scored the best
+     *  *) when multiple contigs are yield, it means the contig has several top-scored configurations
+     * How to handle the second scenario can be treated in a separate logic unit.
+     */
+    @VisibleForTesting
+    static JavaRDD<AlignedContig> filterAndSplitGappedAI(final JavaRDD<AlignedContig> parsedContigAlignments) {
+
+        final Set<String> canonicalChromosomes = getCanonicalChromosomes();
+
+        return parsedContigAlignments
+                .mapToPair(alignedContig -> new Tuple2<>(alignedContig.contigName,
+                        new Tuple2<>(alignedContig.contigSequence, pickBestConfigurations(alignedContig, canonicalChromosomes))))
+                .flatMap(InternalFilterLongReadAlignmentsSAMSpark::reConstructContigFromPickedConfiguration)
+                .sortBy(tig -> tig.contigName, true, parsedContigAlignments.getNumPartitions());
```

Comment:
I run the code with this line commented out, but the output is scrambled...
But I move the sort to clint code.